curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md

```diff
@@ -17,439 +17,445 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the main purpose of using custom properties in CSS?
 
 #### --distractors--
 
-Placeholder distractor 1
+To change the DOM structure
 
 ---
 
-Placeholder distractor 2
+To create dynamic selectors
 
 ---
 
-Placeholder distractor 3
+To improve accessibility
 
 #### --answer--
 
-Placeholder answer
+To define reusable styles across a project
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Custom properties defined in the `:root` selector are considered as?
 
 #### --distractors--
 
-Placeholder distractor 1
+Locally scoped
 
 ---
 
-Placeholder distractor 2
+Non-inheritable
 
 ---
 
-Placeholder distractor 3
+Element-specific
 
 #### --answer--
 
-Placeholder answer
+Globally scoped
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How do you apply a previously defined custom property `--foreground` in CSS?
 
 #### --distractors--
 
-Placeholder distractor 1
+`color: --foreground;`
 
 ---
 
-Placeholder distractor 2
+`color: css(--foreground);`
 
 ---
 
-Placeholder distractor 3
+`color: $foreground;`
 
 #### --answer--
 
-Placeholder answer
+`color: var(--foreground);`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What happens if you use a custom property that is not defined in the current scope?
 
 #### --distractors--
 
-Placeholder distractor 1
+It results in an error
 
 ---
 
-Placeholder distractor 2
+It uses the value `inherit`
 
 ---
 
-Placeholder distractor 3
+The property is ignored
 
 #### --answer--
 
-Placeholder answer
+The browser applies the default value, if provided
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the `@property` rule in CSS?
 
 #### --distractors--
 
-Placeholder distractor 1
+To define a media query
 
 ---
 
-Placeholder distractor 2
+To control CSS animations
 
 ---
 
-Placeholder distractor 3
+To group CSS properties
 
 #### --answer--
 
-Placeholder answer
+To define how custom properties behave
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What should you be cautious of when using custom properties in terms of browser compatibility?
 
 #### --distractors--
 
-Placeholder distractor 1
+Custom properties are automatically converted to `px` units
 
 ---
 
-Placeholder distractor 2
+Custom properties always override inline styles
 
 ---
 
-Placeholder distractor 3
+Custom properties can cause layout shifts
 
 #### --answer--
 
-Placeholder answer
+Older browsers may not support custom properties
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is it important to use fallback values when applying custom properties in CSS?
 
 #### --distractors--
 
-Placeholder distractor 1
+To reduce code complexity
 
 ---
 
-Placeholder distractor 2
+To improve performance
 
 ---
 
-Placeholder distractor 3
+To optimize rendering time
 
 #### --answer--
 
-Placeholder answer
+To avoid undefined values in unsupported browsers
```

Comment:
if the browser doesn't support variables you need a browser fallback, ie the same property above that doesn't use `var()`. See https://www.freecodecamp.org/learn/responsive-web-design/basic-css/improve-compatibility-with-browser-fallbacks for an example. Is this the type of fallback you are asking about?

Reply:
Yes, that’s the type of fallback i mean. Providing a standard property value alongside the CSS variable ensures that the styles work in browsers that don't support variables.

Reply:
it's a fallback also the second value in the `var()` function, maybe a different wording of the question would be more clear?

Reply:
okay on it

Reply:
What is the purpose of providing a fallback value in the var() function for CSS custom properties?

is this right?

Reply:
so you are changing the type of fallback you are asking about? if so make sure to also change the answer

Reply:
done please check